lib/webmachine.ml

```diff
@@ -478,6 +486,12 @@ module Make(IO:IO) = struct
         | `Basic realm ->
           self#set_response_header "WWW-Authenticate" ("Basic realm=\"" ^ realm ^ "\"");
           self#halt 401
+        | `Unauthorized auth ->
+          self#set_response_header "WWW-Authenticate" (String.concat " " (
+              auth.scheme ::
+              ("realm=\"" ^ auth.realm ^ "\"") ::
+              (List.map (fun (k, v) -> k ^ "=\"" ^ v ^ "\"") auth.params)));
```

Comment:
This code could benefit from the use of the [`Buffer`](https://caml.inria.fr/pub/docs/manual-ocaml/libref/Buffer.html) to minimize allocations.